plugins/xml-adapter/src/main/java/org/polypheny/db/adapter/xml/XmlToPolyConverter.java

```diff
@@ -0,0 +1,124 @@
+import javax.xml.stream.XMLStreamConstants;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+import org.apache.commons.codec.binary.Hex;
+import org.polypheny.db.type.entity.PolyBinary;
+import org.polypheny.db.type.entity.PolyBoolean;
+import org.polypheny.db.type.entity.PolyList;
+import org.polypheny.db.type.entity.PolyNull;
+import org.polypheny.db.type.entity.PolyString;
+import org.polypheny.db.type.entity.PolyValue;
+import org.polypheny.db.type.entity.document.PolyDocument;
+import org.polypheny.db.type.entity.numerical.PolyBigDecimal;
+import org.polypheny.db.type.entity.numerical.PolyDouble;
+import org.polypheny.db.type.entity.numerical.PolyFloat;
+import org.polypheny.db.type.entity.numerical.PolyLong;
+import org.polypheny.db.type.entity.relational.PolyMap;
+import org.polypheny.db.type.entity.temporal.PolyDate;
+import org.polypheny.db.type.entity.temporal.PolyTime;
+import org.polypheny.db.type.entity.temporal.PolyTimestamp;
+
+public class XmlToPolyConverter {
+
+    public PolyDocument nodeToPolyDocument( XMLStreamReader reader ) throws XMLStreamException {
+        return new PolyDocument( nodeToPolyMap( reader ) );
+    }
+
+
+    public PolyMap<PolyString, PolyValue> nodeToPolyMap( XMLStreamReader reader ) throws XMLStreamException {
+        Map<PolyString, PolyValue> map = new HashMap<>();
+        String currentElement = reader.getLocalName(); // Root element of the current document
+        while ( reader.hasNext() ) {
+            int event = reader.next();
+            if ( event == XMLStreamConstants.START_ELEMENT ) {
+                PolyString key = new PolyString( reader.getLocalName() );
+                PolyValue value = nodeToPolyValue( reader );
+                map.put( key, value );
+            } else if ( event == XMLStreamConstants.END_ELEMENT && reader.getLocalName().equals( currentElement ) ) {
+                break; // Stop processing at the end of the current document element
+            }
+        }
+        return PolyMap.of( map );
+    }
+
+
+    public PolyValue nodeToPolyValue( XMLStreamReader reader ) throws XMLStreamException {
+        if ( reader.getEventType() == XMLStreamConstants.START_ELEMENT ) {
+            String type = reader.getAttributeValue( null, "type" );
+            return convertByType( reader, type != null ? type : "string" );
+        }
+        return new PolyNull(); // Default case for unsupported nodes
+    }
+
+
+    private PolyValue convertByType( XMLStreamReader reader, String type ) throws XMLStreamException {
+        StringBuilder value = new StringBuilder();
+        while ( reader.hasNext() ) {
+            int event = reader.next();
+            if ( event == XMLStreamConstants.CHARACTERS ) {
+                value.append( reader.getText().trim() );
+            } else if ( event == XMLStreamConstants.END_ELEMENT ) {
+                break;
+            }
+        }
+
+        if ( value.length() == 0 ) {
```

Comment:
Can be `value.isEmpty()`.

Reply:
This class received huge changes. This line thus no longer exists..